web/modules/custom/custom_weather/src/Form/CustomWeatherUserForm.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Drupal\custom_weather\Form;
+
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\custom_weather\Service\UserCityHandler;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines a configuration form for managing settings related to the module.
+ */
+class CustomWeatherUserForm extends FormBase {
+
+  /**
+   * Stores current user city.
+   */
+
+  protected mixed $city;
+
+  /**
+   * Stores current user id.
+   */
+  protected int $currentUser;
+
+  /**
+   * Stores DB connection.
+   */
+  protected mixed $connection;
```

Comment:
As of PHP 8.0.0, constructor parameters may also be promoted to correspond to an object property. It is very common for constructor parameters to be assigned to a property in the constructor but otherwise not operated upon. Constructor promotion provides a short-hand for that use case. The example above could be rewritten as the following.

When a constructor argument includes a modifier, PHP will interpret it as both an object property and a constructor argument, and assign the argument value to the property. The constructor body may then be empty or may contain other statements. Any additional statements will be executed after the argument values have been assigned to the corresponding properties.

Not all arguments need to be promoted. It is possible to mix and match promoted and not-promoted arguments, in any order. Promoted arguments have no impact on code calling the constructor.